test/github-issues/2588/issue-2588.ts

```diff
@@ -0,0 +1,47 @@
+import "reflect-metadata";
+import {Connection} from "../../../src/connection/Connection";
+import {closeTestingConnections, createTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {expect} from "chai";
+
+import {Post} from "./entity/Post";
+import {PostReview} from "./entity/PostReview";
+
+describe("github issues > #2588 - createQueryBuilder always does left joins on relations", () => {
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        enabledDrivers: ["postgres"],
```

Comment:
Bug was reported for postgres, but doesn't it occur on every database using SelectQueryBuilder? It doesn't look like database engine related. If that's the case we should probably test it on every driver.